sjsonnet/test/src/sjsonnet/UnicodeHandlingTests.scala

```diff
@@ -167,23 +209,32 @@ object UnicodeHandlingTests extends TestSuite {
 
       // JSON manifest variants:
       eval(s"std.manifestJsonMinified($testObject)") ==>
-      ujson.Str("{\"a\":1,\"z\":2,\"\uFFFF\":3,\"\uD800\uDC00\":4}")
+        ujson.Str("{\"a\":1,\"z\":2,\"\uFFFF\":3,\"\uD800\uDC00\":4}")
 
       eval(s"std.manifestJson($testObject)") ==>
-      ujson.Str("{\n    \"a\": 1,\n    \"z\": 2,\n    \"\uFFFF\": 3,\n    \"\uD800\uDC00\": 4\n}")
+        ujson.Str("{\n    \"a\": 1,\n    \"z\": 2,\n    \"\uFFFF\": 3,\n    \"\uD800\uDC00\": 4\n}")
 
       eval(s"std.manifestJsonEx($testObject, '  ')") ==>
-      ujson.Str("{\n  \"a\": 1,\n  \"z\": 2,\n  \"\uFFFF\": 3,\n  \"\uD800\uDC00\": 4\n}")
+        ujson.Str("{\n  \"a\": 1,\n  \"z\": 2,\n  \"\uFFFF\": 3,\n  \"\uD800\uDC00\": 4\n}")
 
       // TOML manifest:
       eval(s"std.manifestTomlEx($testObject, '  ')") ==>
-      ujson.Str("a = 1\nz = 2\n\"\\uffff\" = 3\n\"\\ud800\\udc00\" = 4")
+        ujson.Str("a = 1\nz = 2\n\"\\uffff\" = 3\n\"\\ud800\\udc00\" = 4")
     }
 
     test("findSubstr") {
       eval("std.findSubstr('🌍', 'Hello 🌍 World')") ==> ujson.Arr(6)
       eval("std.findSubstr('o', 'Hello 🌍 World')") ==> ujson.Arr(4, 9)
       eval("std.findSubstr('🌍🚀', '🌍🚀 and more 🌍🚀')") ==> ujson.Arr(0, 12)
     }
+
+    test("mixedUnicodeAndAscii") {
+      // Test mixing ASCII and Unicode characters
+      eval("'Hello ' + '🌍' + ' World'") ==> ujson.Str("Hello 🌍 World")
+      eval("std.length('ASCII🌍混合')") ==> ujson.Num(8)
+      eval("'ASCII🌍混合'[5]") ==> ujson.Str("🌍")
+      eval("'ASCII🌍混合'[6]") ==> ujson.Str("混")
+      eval("'ASCII🌍混合'[7]") ==> ujson.Str("合")
+    }
```

Comment:
What coverage is this new test case adding? I think that slicing and length of mixed ASCII and Unicode strings are both already covered by existing tests. Perhaps the new thing here is a testing of string concatenation?

Reply:
Testing the `indexing` but it should be covered in other place